server/src/main/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldMapper.java

```diff
@@ -400,33 +469,21 @@ public void parseKnnVectorAndIndex(DocumentParserContext context, DenseVectorFie
             @Override
             double parseKnnVectorToByteBuffer(DocumentParserContext context, DenseVectorFieldMapper fieldMapper, ByteBuffer byteBuffer)
                 throws IOException {
-                double dotProduct = 0f;
-                int index = 0;
-                for (Token token = context.parser().nextToken(); token != Token.END_ARRAY; token = context.parser().nextToken()) {
-                    fieldMapper.checkDimensionExceeded(index, context);
-                    ensureExpectedToken(Token.VALUE_NUMBER, token, context.parser());
-                    int value = context.parser().intValue(true);
-                    if (value < Byte.MIN_VALUE || value > Byte.MAX_VALUE) {
-                        throw new IllegalArgumentException(
-                            "element_type ["
-                                + this
-                                + "] vectors only support integers between ["
-                                + Byte.MIN_VALUE
-                                + ", "
-                                + Byte.MAX_VALUE
-                                + "] but found ["
-                                + value
-                                + "] at dim ["
-                                + index
-                                + "];"
-                        );
-                    }
-                    byteBuffer.put((byte) value);
-                    dotProduct += value * value;
-                    index++;
-                }
-                fieldMapper.checkDimensionMatches(index, context);
-                return dotProduct;
+                XContentParser.Token token = context.parser().currentToken();
+                return switch (token) {
+                    case START_ARRAY -> parseVectorArray(context, fieldMapper, (val, idx) -> byteBuffer.put(val));
+                    case VALUE_STRING -> parseHexEncodedVector(context, fieldMapper, (val, idx) -> byteBuffer.put(val));
+                    case VALUE_NUMBER -> parseNumberVector(context, fieldMapper, (val, idx) -> byteBuffer.put(val));
```

Comment:
We shouldn't allow this. If this is something we want to support (just a number that gets put into a vector of dim==1), then it should be a separate PR. Personally, I am against it as it encourages bad behavior. If folks have a single value, they should index it as keyword or `int` or `float`

Reply:
I'm definitely with you on this, but added this because it is what we already have and did not want to change the existing API. 

Currently, when parsing the request we make use of `declareFloatArray` which ends up being defined as 
```
        FLOAT_ARRAY(START_ARRAY, VALUE_NUMBER, VALUE_STRING)
```
hence, while far from ideal, we already support single valued numbers. I'm +1 if you agree to remove this (don't think that it'd actually affect anyone tbf)

Reply:
> hence, while far from ideal, we already support single valued numbers

```
POST vectors/_doc
{
	"vector": 1
}
```

```
"caused_by": {
            "type": "parsing_exception",
            "reason": "Failed to parse object: expecting token of type [VALUE_NUMBER] but found [END_OBJECT]",
            "line": 4,
            "col": 1
        }
```

When indexing.

But we do allow it on the query side. So, lets disallow it on indexing, but continue to allow it query side.

Reply:
Guess I missed that :/ Somehow I was under the impression that this was consistently allowed on both indexing & searching. Will proceed to disallow it on indexing and keep the existing behavior on query-time.